desc/plotting.py

```diff
@@ -2083,7 +2090,17 @@ def plot_boundaries(
         fig, ax = plot_boundaries((eq1, eq2, eq3))
 
     """
+    # if NFPs are not all equal, means there are
+    # objects with differing NFPs, which it is not clear
+    # how to choose the phis for by default, so we will throw an error
+    # unless phi was given.
     phi = parse_argname_change(phi, kwargs, "zeta", "phi")
+    errorif(
+        not np.allclose([thing.NFP for thing in eqs], eqs[0].NFP) and phi is None,
+        ValueError,
+        "supplied objects must have the same number of field periods, "
+        "or if there are differing field periods, `phi` must be given explicitly.",
```

Comment:
Can we add the supplied NFPs in the error message as we usually do i.e. "same nfp expected but got {[thing.NFP for thing in eqs]}"? Also, maybe also mentioning about axisymmetric case is better since that is a more common case (but with this pr we dont let it)